File: template-method/typescript/xml-data-analyzer.ts
import {DataAnalyzerBase} from "./data-analyzer-base";

// UML diyagramındaki AbstractClass (DataAnalyzerBase) sınıfından türer.
// UML diyagramındaki ConcreteClass yapısına denk gelmektedir.
export class XmlDataAnalyzer extends DataAnalyzerBase {
  public exportResult(result: object): void {
    // Durumun XML'e dönüştürülmesi işlemleri yer alacaktır.
    console.log("XML exported.");
  }
}

File: template-method/go/xml-data-analyzer.go
package main

import "fmt"

// UML diyagramındaki AbstractClass (DataAnalyzerBase) sınıfından türer.
// UML diyagramındaki ConcreteClass yapısına denk gelmektedir.
type xmlDataAnalyzer struct {
	baseAnalyzer *DataAnalyzerBase
}

func NewXMLDataAnalyzer(b *DataAnalyzerBase) *xmlDataAnalyzer {
	return &xmlDataAnalyzer{
		baseAnalyzer: b,
	}
}

func (h xmlDataAnalyzer) exportResult(_ []byte) {
	// Durumun XML'e dönüştürülmesi işlemleri yer alacaktır.e
	fmt.Println("XML exported.")
}
